src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final double GROW_VALUE = 1.5;
+    private int maxSize = 10;
+    private int actualSize = 0;
+    private Object[] array = new Object[maxSize];
+    private Object[] tempArray;
+
     @Override
     public void add(T value) {
-
+        if (actualSize == maxSize) {
+            grow();
+        }
+        array[actualSize] = value;
+        actualSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > actualSize || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist");
+        }
+        if (actualSize == maxSize) {
+            grow();
+        }
+        tempArray = new Object[maxSize];
+        System.arraycopy(array, 0, tempArray, 0, index);
```

Comment:
in each method of this class, use arraycopy only once, you can pass the same array as both src and dest